common/src/main/java/com/microsoft/identity/common/internal/cache/ISaveTokenResult.java

```diff
@@ -0,0 +1,63 @@
+// Copyright (c) Microsoft Corporation.
+// All rights reserved.
+//
+// This code is licensed under the MIT License.
+//
+// Permission is hereby granted, free of charge, to any person obtaining a copy
+// of this software and associated documentation files(the "Software"), to deal
+// in the Software without restriction, including without limitation the rights
+// to use, copy, modify, merge, publish, distribute, sublicense, and / or sell
+// copies of the Software, and to permit persons to whom the Software is
+// furnished to do so, subject to the following conditions :
+//
+// The above copyright notice and this permission notice shall be included in
+// all copies or substantial portions of the Software.
+//
+// THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+// IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+// FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+// AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+// LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+// OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN
+// THE SOFTWARE.
+package com.microsoft.identity.common.internal.cache;
+
+import com.microsoft.identity.common.internal.dto.AccessToken;
+import com.microsoft.identity.common.internal.dto.Account;
+import com.microsoft.identity.common.internal.dto.IdToken;
+import com.microsoft.identity.common.internal.dto.RefreshToken;
+
+/**
+ * Result container for Account and Credential - usually the result of a save or load operation.
+ */
+public interface ISaveTokenResult {
+
+    /**
+     * Gets the {@link Account}.
+     *
+     * @return The Account to get.
+     */
+    Account getAccount();
+
+    /**
+     * Gets the {@link AccessToken}.
+     *
+     * @return The AccessToken to get.
+     */
+    AccessToken getAccessToken();
```

Comment:
@shoatman Would it be preferable to just return a `List<Credential>` and forget the various `Credential` subtypes entirely? (Thinking about x-IdP compat, sanity)

Reply:
Discussed offline, going to leave as-is